File: 🟡 Medium/386 - Lexicographical Numbers/_386LexicographicalNumbers.py
class Solution:
    def dfs(self, i: int, n: int, res: List[int]):
        if i > n:
            return

        res.append(i)

        for j in range(0, 10):
            self.dfs(i * 10 + j, n, res)

    def lexicalOrder(self, n: int) -> List[int]:
        res = []
        for i in range(1, 10):
            self.dfs(i, n, res)
        
        return res

File: 🟡 Medium/386 - Lexicographical Numbers/_386LexicographicalNumbers.java
class Solution {
    public void dfs(int i, int n, List<Integer> res) {
        if (i > n) {
            return;
        }

        res.add(i);

        for (int j = 0; j < 10; j++) {
            dfs(10 * i + j, n, res);
        }
    }

    public List<Integer> lexicalOrder(int n) {
        List<Integer> res = new ArrayList<>();
        for (int i = 1; i <= 9; i++) {
            dfs(i, n, res);
        }

        return res;
    }
}